_posts/2023-08-17-Announcing-Data-Prepper-2.4.0.md

```diff
@@ -0,0 +1,285 @@
+```
+
+See the documentation for complete set of [configurable options](https://opensearch.org/docs/latest/data-prepper/pipelines/configuration/sources/s3/).
+
+
+## Filtering in sinks
+
+User can configure the option of “include_keys” and “exclude_keys” under sink to allow or deny data being written to sinks. 
+This provides flexibility to ingest data from any source, apply common enrichments using a processor chain and selective send data to a specific sink like OpenSearch and send all other data to sink like S3 for archival purpose. 
+Archived data in S3 can be written in Parquet so it can be easily be queried by any other compatible engine. 
+Users can also then leverage S3 batch processing to hydrate archived data back to OpenSearch if needed for richer analytics. 
+Such filtering enables you to optimize your storage and reduce costs by sending specific data to specific sinks.
+
+```
+sink:
+- opensearch:
+  include_keys:
+  - srcport
+  - dstport
+  - srcaddr
+  - dstaddr
+  - start
+  - end
+
+- s3:
+  exclude_keys:
+  - version
+  - interfaceId
+```
+
+
+S3 sink codecs
+
+Data Prepper 2.4.0 adds useful codecs like Avro and Parquet to the s3 sink to facilitate more use cases. This release includes four new codecs for the data it is writing.
+
+* JSON codec - This codec will write an S3 object which is a valid JSON object. All the events will be found within an array. This can be useful when integrating with systems that expect a JSON object.
+* Avro codec - You can define your Avro schema and write events in the efficient [Avro format](https://avro.apache.org/docs/).
+* Parquet codec - [Apache Parquet](https://parquet.apache.org/) is a popular columnar format. By saving your Data in Parquet format you can later efficiently retrieve data for analysis. You can define your schema in Avro format for Parquet codec as well.
+* CSV codec - This codec supports writing events as a delimited format, this can be CSV, TSV, or other delimited formats.
+
+
+For example, you can configure the Avro codec with a schema for network traffic using a similar approach to the following.
+
+
+```
+s3:
+  codec:
+    avro:
+      schema: >
+        {
+          "type" : "record",
+          "namespace" : "org.opensearch.dataprepper.examples",
+          "name" : "NetworkTraffic",
+          "fields" : [
+            { "name" : "sourcePort", "type": "int"},
+            { "name" : "destinationPort", "type": "int"},
+            { "name" : "sourceAddress", "type" : "string"},
+            { "name" : "destinationAddress", "type" : "string"},
+            { "name" : "bytes", "type": "int"},
+          ]
+        }
+```
+
+Additionally, the S3 sink can now write compressed gzip or compressed Snappy files to reduce your network and storage needs. 
+You can include this in your S3 sink configuration similar to the following.
+
+```
+s3:
+  compression: gzip # Other option is "snappy"
+```
+
+
+
+## Streaming anomaly detection with high cardinality
```

Comment:
Rewrite here:

--- 

The streaming `anomaly_detector` now contains the `identification_keys` options, which creates a Random Cut Forest (RCF) model for each value in the your time-series data. With the `identification_keys` option, anomalies can be detected in unique set of keys.

The following examples create an `anomaly_detection` processor that detects anomalies for each unique IP address:

```json
processor:
- anomaly_detector:
   identification_keys: ["ip"]
   keys: ["latency"]
   mode:
    random_cut_forest:
```

You can write anomalies detected by the processor to a seperate index and create alerts using [document level monitors](https://opensearch.org/docs/latest/observing-your-data/alerting/api/#document-level-monitors).

Furthermore, you can use the `verbose` configuration in the anomaly detection processor to change the number of alerts and anomalies shown by the processor, `false` for less, `true` for more.